abodepy/event_controller.py

```diff
@@ -46,6 +50,36 @@ def stop(self):
         """Tell the subscription thread to terminate - will block."""
         self._socketio.stop()
 
+    def add_connection_status_callback(self, devices, callback):
+        """Register callback for Abode server connection status."""
+        if not devices:
+            return False
+
+        if not isinstance(devices, (tuple, list)):
+            devices = [devices]
+
+        for device in devices:
+            device_id = device
+
+            if isinstance(device, AbodeDevice):
+                device_id = device.device_id
+
+                if not self._abode.get_device(device_id):
+                    raise AbodeException((ERROR.EVENT_DEVICE_INVALID))
+
+            if isinstance(device, AbodeAutomation):
+                device_id = device.automation_id
+
+                if not self._abode.get_automation(device_id):
+                    raise AbodeException((ERROR.EVENT_DEVICE_INVALID))
+
+            _LOGGER.debug(
+                "Subscribing to Abode connection updates for device_id: %s", device_id)
```

Comment:
At a minimum I'll change this to `"Subscribing to Abode connection updates for: %s", device_id)` (remove device_id from the string).